### PR TITLE
Made changes in the ParseOptionTickerOSI function. Now it can handle …

### LINKED ISSUE #8205: GenerateOptionTickerOSI/ParseOptionTickerOSI Improvement
#### Expected Behavior

- `GenerateOptionTickerOSI/ParseOptionTickerOSI` at `SymbolRepresentation` can handle no empty spaces between underlying and expiration date for example: `AAPL240614C00100000`
#### Actual Behavior

- It current expected `AAPL  240614C00100000`
#### Potential Solution

- Improve implementation to support both formats
- Will allow us to clean up Alpaca symbol mapper by reusing lean helper methods instead  see https://github.com/QuantConnect/Lean.Brokerages.Alpaca/blob/master/QuantConnect.AlpacaBrokerage/AlpacaBrokerageSymbolMapper.cs#L48
#### Reproducing the Problem

N/A
#### System Information

N/A
#### Checklist

- [x] I have completely filled out this template
- [x] I have confirmed that this issue exists on the current `master` branch
- [x] I have confirmed that this is not a duplicate issue by searching [issues](https://github.com/QuantConnect/Lean/issues)

- [x] I have provided detailed steps to reproduce the issue

### PR DESCRIPTION
**Pull request by keshavdalmia10**
…both with and without space

#### Description

Closes #8205 

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Refactor (non-breaking change which improves implementation)
- [x] Performance (non-breaking change which improves performance. Please add associated performance test and results)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:

- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [x] I have added tests to cover my changes. 
- [x] All new and existing tests passed.
- [x] My branch follows the naming convention `bug-<issue#>-<description>` or `feature-<issue#>-<description>`

### CODE REVIEW

**Comment by keshavdalmia10** [comment]
Just changed the branch name right now. Will make the necessary changes as you requested.
Thanks

**Comment by keshavdalmia10** [comment]
Hi @Martin-Molinero , I have made the changes you told. If any further changes required feel free to tell. 
Thanks.

File: Common/SymbolRepresentation.cs [hunk 1]
```diff
@@ -316,6 +320,37 @@ public static string GenerateOptionTickerOSI(string underlying, OptionRight righ
...
+        /// More information can be found at http://www.optionsclearing.com/components/docs/initiatives/symbology/symbology_initiative_v1_8.pdf
+        /// </summary>
+        /// <param name="underlying">Underlying string</param>
+        /// <param name="right">Option right</param>
+        /// <param name="strikePrice">Option strike</param>
+        /// <param name="expiration">Option expiration date</param>
+        /// <returns>The OSI ticker representation</returns>
+        public static string GenerateOptionTickerOSICompact(string underlying, OptionRight right, decimal strikePrice, DateTime expiration)
+        {
+        return Invariant($"{underlying}{expiration.ToStringInvariant(DateFormat.SixCharacter)}{right.ToStringPerformance()[0]}{(strikePrice * 1000m):00000000}");
```
Comment (Martin-Molinero) [inline Common/SymbolRepresentation.cs:351]: Indentation is wrong for the lines in these 2 new methods? missing 1 tab/level deeper it seems

File: Tests/Common/SymbolRepresentationTests.cs [hunk 2]
```diff
@@ -53,6 +67,25 @@ public void OptionSymbolAliasAddsPaddingSpaceForSixOrMoreCharacterSymbols()
...
+        [TestCase("MSFT", OptionRight.Put, 150.00, "240614", "MSFT240614P00150000")]
+        [TestCase("GOOG", OptionRight.Call, 2000.00, "211231", "GOOG211231C02000000")]
+        [TestCase("AMZN", OptionRight.Put, 3500.50, "250101", "AMZN250101P03500500")]
+        [TestCase("NFLX", OptionRight.Call, 500.25, "221201", "NFLX221201C00500250")]
+        [TestCase("TSLA", OptionRight.Put, 725.00, "241231", "TSLA241231P00725000")]
+        [TestCase("V", OptionRight.Call, 220.00, "230420", "V230420C00220000")]
+        [TestCase("JPM", OptionRight.Put, 130.75, "230710", "JPM230710P00130750")]
+        [TestCase("IBM", OptionRight.Call, 145.00, "250212", "IBM250212C00145000")]
+        [TestCase("BABA", OptionRight.Put, 88.88, "240508", "BABA240508P00088880")]
+        public void GenerateOptionTickerOSICompact_ValidInputs(string underlying, OptionRight right, decimal strikePrice, string date, string expected)
```
Comment (Martin-Molinero) [inline Tests/Common/SymbolRepresentationTests.cs:82]: Nice!

File: Tests/Common/SymbolRepresentationTests.cs [hunk 3]
```diff
@@ -34,6 +34,20 @@ public void OptionSymbolAliasMatchesOSI()
...
+        [TestCase("V     231211P00220000", SecurityType.Option, OptionStyle.American, "V", "V")]
+        [TestCase("JPM   240501C00130750", SecurityType.Option, OptionStyle.American, "JPM", "JPM")]
+        [TestCase("IBM   250212P00145000", SecurityType.Option, OptionStyle.American, "IBM", "IBM")]
+        [TestCase("BABA  240508C00088880", SecurityType.Option, OptionStyle.American, "BABA", "BABA")]
+        [TestCase("DIS   230630C00075000", SecurityType.Option, OptionStyle.American, "DIS", "DIS")]
+        [TestCase("ORCL  231030C00065000", SecurityType.Option, OptionStyle.American, "ORCL", "ORCL")]
+        [TestCase("CSCO  230501P00045000", SecurityType.Option, OptionStyle.American, "CSCO", "CSCO")]
+        [TestCase("DAX   250715C01000000", SecurityType.IndexOption, OptionStyle.European, "DAX", "DAX")]
+        [TestCase("FTSE  230122C00750000", SecurityType.IndexOption, OptionStyle.European, "FTSE", "FTSE")]
         public void ParseOptionTickerOSI(string optionStr, SecurityType securityType, OptionStyle optionStyle,
```
Comment (Martin-Molinero) [inline Tests/Common/SymbolRepresentationTests.cs:51]: Let's please add assertion on the expected `decimal strikePrice, string date` as done below 👍

**Review by Martin-Molinero — CHANGES_REQUESTED** [review]
Thanks! almost there 👍

**Comment by keshavdalmia10** [comment]
Hi @Martin-Molinero, I have made the requested changes.
Thanks

**Comment by Martin-Molinero** [comment]
Thank you @keshavdalmia10! Nice work 🚀

**Pull request merged**